cmd/create/cluster/cmd.go

```diff
@@ -3079,34 +3101,39 @@ func run(cmd *cobra.Command, _ []string) {
 
 	if isSTS {
 		if mode != "" {
-			if !output.HasFlag() || r.Reporter.IsTerminal() {
-				r.Reporter.Infof("Preparing to create operator roles.")
-			}
-			err := operatorroles.Cmd.RunE(operatorroles.Cmd, []string{clusterName, mode, permissionsBoundary})
-			if err != nil {
-				r.Reporter.Errorf("There was a problem creating operator roles: %v", err)
-				os.Exit(1)
+			if len(operatorRoles) == 0 {
+				if !output.HasFlag() || r.Reporter.IsTerminal() {
+					r.Reporter.Infof("Preparing to create operator roles.")
+				}
+				err := operatorroles.Cmd.RunE(operatorroles.Cmd, []string{clusterName, mode, permissionsBoundary})
+				if err != nil {
+					r.Reporter.Errorf("There was a problem creating operator roles: %v", err)
+					os.Exit(1)
+				}
 			}
 			if !output.HasFlag() || r.Reporter.IsTerminal() {
 				r.Reporter.Infof("Preparing to create OIDC Provider.")
 			}
 			oidcprovider.Cmd.Run(oidcprovider.Cmd, []string{clusterName, mode, ""})
 		} else {
 			output := ""
-			if cluster.AWS().STS().OidcConfig().Reusable() {
-				output = "When using reusable OIDC Config and resources have been created " +
-					"prior to cluster specification, this step is not required."
+			if len(operatorRoles) == 0 {
+				rolesCMD := fmt.Sprintf("rosa create operator-roles --cluster %s", clusterName)
+				if permissionsBoundary != "" {
+					rolesCMD = fmt.Sprintf("%s --permissions-boundary %s", rolesCMD, permissionsBoundary)
+				}
+				output = fmt.Sprintf("%s\t%s\n", output, rolesCMD)
 			}
-			rolesCMD := fmt.Sprintf("rosa create operator-roles --cluster %s", clusterName)
-			if permissionsBoundary != "" {
-				rolesCMD = fmt.Sprintf("%s --permissions-boundary %s", rolesCMD, permissionsBoundary)
+			if cluster.AWS().STS().OidcConfig() == nil || !cluster.AWS().STS().OidcConfig().Reusable() {
+				oidcCMD := "rosa create oidc-provider"
```

Comment:
A similar treatment might be required here where you first check if the oidc provider is already present in the customer account instead of the oidc config condition

Reply:
How would I check if the provider is already present? I thought the call here [handleOidcConfigOptions()](https://github.com/openshift/rosa/blob/master/cmd/create/cluster/cmd.go#L1664) was checking and if provider was present (or entered) then it would be added to the cluster.

Or do you mean I should make the call that `rosa list oidc-provider` is making? If so how do I know which of the providers returned would satisfy the need not to have to run `rosa create oidc-provider`? The call takes a _very_ long time for me (48s), if that matters.

Reply:
This call requires the oidcConfig.IssuerUrl() but that is nil, right?

Reply:
Shouldn't be nil after the cluster create request is through

Reply:
thanks!